### PR TITLE
Hcal DD Rec and Sim Constants Cleanup

### PR DESCRIPTION
**Pull request by ianna**
- Pick up vectors directly from HcalParameters record

### CODE REVIEW

**Comment by ianna** [comment]
@bsunanda - Please, check this. I've removed a few locally cached vectors and replaced them with directly accessed from HcalParameters record. Thanks.

File: Geometry/HcalCommonData/interface/HcalDDDRecConstants.h [hunk 1]
```diff
@@ -80,29 +80,18 @@ class HcalDDDRecConstants {
...
-  int                 modeTopo_;  // Mode for topology
-  std::vector<int>    etaGroup;   // Eta Grouping
   std::vector<std::pair<int,int> > etaSimValu; // eta ranges at Sim stage
-  std::vector<double> etaTable;   // Eta table (HB+HE)
   std::vector<int>    ietaMap;    // Map Sim level ieta to Rec level ieta
-  std::vector<int>    iEtaMin, iEtaMax; // Minimum and maximum eta
-  std::vector<int>    maxDepth;   // Maximum depth in HB/HE/HF/HO           
-  int                 nEta;       // Number of bins in eta for HB and HE
-  std::vector<int>    phiGroup;   // Eta Grouping
   std::vector<double> phibin;     // Phi step for all eta bins (HB, HE, HO)
```
Comment (ianna) [inline Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: @bsunanda - would it make sense to add the remaining vectors which are calculated at initialisation of this class to HcalParameters? The same for Sim part?

Comment (bsunanda) [inline Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: I dont know which could be persistent and which could be transient variables. This depends on the analysis if you could make some computation at the beginning and use them later on. At this stage I suggest please do not change the payload and whatever can be calculated may be there at initialization.
________________________________

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by bsunanda** [comment]
Dear Yana

I am reverting back some of the changes and keeping most of the ones which are surely improvements. I shall commit the changes sometime today and please see that they get integrated quickly enough. I am running out of time for pre2. I want all geometry related stuff to go in by pre2. Best regards

Sunanda
________________________________

**Pull request merged**